module_1/src/components/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { FocusTrap } from 'focus-trap-react';
+import { Portal } from 'react-portal';
+import styles from './Dialog.module.css';
+
+interface DialogProps {
+  title: string | JSX.Element;
+  children: React.ReactNode;
+  onClose: () => void;
+}
+
+export const Dialog: React.FC<DialogProps> = ({ title, children, onClose }) => {
+  return (
+    <Portal node={document.getElementById("portal-root")!}>
```

Comment:
Minor:
Potential runtime error if portal-root element doesn't exist
```suggestion
    <Portal node={document.getElementById("portal-root") || document.body}>
```